### ISSUE TITLE
[Proposal] `dapr run` track and `dapr list` display the PID of run commands

### ISSUE THREAD

**Issue by philliphoff** [dapr/cli#1191]
## Describe the proposal

Currently, `dapr list` displays the PIDs of running Dapr sidecars and the CLI that started them (if any).  However, the Dapr CLI also starts commands (i.e. the applications themselves); the PIDs of these commands are not currently tracked.  Tooling used to attach debuggers would be helped if they could more directly determine the PIDs of commands started by the CLI.

For example:

```json
[
  {
    "appId": "vscode-test",
    "httpPort": 64040,
    "grpcPort": 64041,
    "appPort": 3000,
    "metricsEnabled": true,
    "command": "node ./bin/www",
    "age": "7s",
    "created": "2023-01-30 10:32.27",
    "daprdPid": 18398,
    "cliPid": 18392,
    "commandPid": 18399, # <-- New property with PID of command run, if any (null or omitted if not)
    "maxRequestBodySize": -1,
    "httpReadBufferSize": -1,
    "runTemplatePath": "/Users/phoff/Source/scratch/vscode-test/dapr.yaml"
  }
]
```

> Note that, depending on the command, tooling may still need to walk the process tree to get to a particular process (e.g. if using a `npm` script to start a Node application, `go run`, etc.), but this would still provide a better starting point than the CLI process PID.

## Release Note

RELEASE NOTE:

**Comment by philliphoff** [comment]
I think this may be a significant tooling blocker.  The problem is that, in a larger orchestration of similar application types, it's possible that the commands are the same across applications.  For example, two .NET applications might both be started with the command `dotnet run` (in their respective directories).  Since both applications are started by the same parent `dapr` process, it's not straightforward (across platforms) to correlate each child command process to its respective application.

Note, also, that the listed process command may not exactly match the command in the run file.  For example, the run file may use an alias (e.g. `python`) but the actual process listing has the actual linked path, making correlation impossible.

**Comment by philliphoff** [comment]
Note that the PR has the application PID listed as `"appPid"` instead of the original proposed `"commandPid"` as that seemed more consistent with the other terminology used in the codebase.

**Comment by mukundansundar** [comment]
closed via #1200
